packages/compose/src/hooks/use-drop-zone/index.js

```diff
@@ -228,6 +235,6 @@ export default function useDropZone( {
 				);
 			};
 		},
-		[ isDisabled ]
+		[ isDisabled, dropZoneRef, dropZoneRef?.current ] // Refresh when the passed in dropZoneRef changes.
```

Comment:
There's possibly only a need for `dropZoneRef?.current` here?

I think the ref object by definition shouldn't change.

Reply:
Good catch, yes, we only need to look for `dropZoneRef?.current` 👍
Updated in https://github.com/WordPress/gutenberg/pull/50726/commits/4eb834dfffa4bb2cf1d301ffb3095a585ded882b

Reply:
IIRC, putting `ref.current` inside the dependency array should trigger lint warning. Since updating `ref.current` won't trigger a re-render, it doesn't make sense to put that in the dependency array.

Reply:
Thanks 👍
I've pushed a few changes, so we're now passing a `dropZoneElement` param instead of a `ref`, so the dependency array now just has `isDisabled` and `dropZoneElement` in it.

Reply:
Here are some related articles about this:

- https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/
- https://stackoverflow.com/a/60476525

Reply:
Oh nice! I didn't noticed that. Ignore this please 😆 .

Reply:
All good — I appreciate the links and extra knowledge as always! 🤗